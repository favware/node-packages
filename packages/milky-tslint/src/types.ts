/* eslint-disable @typescript-eslint/no-explicit-any */
export type IPluginOptions = {
  configuration?: any;
  fix?: boolean;
  quiet?: boolean;
  formatter?: LintFormatters;
  formattersDirectory?: string;
  rulesDirectory?: string;
  tslint?: any;
  program?: any;
};

export type IReportOptions = {
  emitError?: boolean;
  reportLimit?: number;
  summarizeFailureOutput?: boolean;
  allowWarnings?: boolean;
};

export type ITSLintFile = {
  tslint: any;
  path: string;
  relative: string;
  contents: Buffer | any;

  isStream (): boolean;

  isNull (): boolean;
};

export enum LogLevels {
  ERROR = 'error',
  INFO = 'info',
  WARN = 'warn',
}

export enum LintFormatters {
  CHECKSTYLE = 'checkstyle',
  CODEFRAME = 'codeFrame',
  FILESLIST = 'filesList',
  JSON = 'json',
  JUNIT = 'junit',
  MSBUILD = 'msbuild',
  PMD = 'pmd',
  PROSE = 'prose',
  STYLISH = 'stylish',
  TAP = 'tap',
  VERBOSE = 'verbose',
  VSO = 'vso',
}