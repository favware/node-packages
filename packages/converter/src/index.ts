export * from './converter';
export * from './helpers';
