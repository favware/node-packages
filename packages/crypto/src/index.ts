import crypto from './crypto';

export * from './util';
export * from './crypto';
export * from './StreamBinary';
export default crypto;
