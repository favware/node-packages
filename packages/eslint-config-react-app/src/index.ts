/**
 * Standard ESLint config for Favware projects using React + TypeScript
 * @example
 * ```json
 * {
 *   "extends": "@favware/eslint-config-react-app"
 * }
 * ```
 */
export const config = {
  root: true,
  extends: ['@favware', 'plugin:import/typescript'],
  plugins: ['import', 'jsx-a11y', 'react', 'react-hooks'],
  parserOptions: {
    project: './tsconfig.eslint.json',
    sourceType: 'module',
    ecmaVersion: 2019,
    ecmaFeatures: { jsx: true },
    jsx: true
  },
  settings: {
    react: {
      version: 'detect'
    }
  },
  rules: {
    // Overwriting config from @favware/eslint-config to better suite React applications
    'no-invalid-this': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/naming-convention': [
      'error',
      {
        selector: 'default',
        format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
        leadingUnderscore: 'forbid',
        trailingUnderscore: 'forbid',
        filter: {
          regex: '^(?:VariableDeclarator|AssignmentExpression)$',
          match: false
        }
      },
      {
        selector: 'parameter',
        format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
        leadingUnderscore: 'allow'
      },
      {
        selector: 'property',
        modifiers: ['readonly'],
        format: ['PascalCase']
      },
      {
        selector: 'enumMember',
        format: ['PascalCase']
      },
      {
        selector: 'typeLike',
        format: ['PascalCase']
      }
    ],

    // Implement desired rules that also get set by eslint-config-react-app
    'no-extra-label': 'warn',
    'no-iterator': 'warn',
    'no-label-var': 'warn',
    'no-labels': ['warn', { allowLoop: true, allowSwitch: true }],
    'no-lone-blocks': 'warn',
    'no-loop-func': 'warn',
    'no-mixed-operators': [
      'warn',
      {
        groups: [
          ['&', '|', '^', '~', '<<', '>>', '>>>'],
          ['==', '!=', '===', '!==', '>', '>=', '<', '<='],
          ['&&', '||'],
          ['in', 'instanceof']
        ],
        allowSamePrecedence: false
      }
    ],
    'no-multi-str': 'warn',
    'rest-spread-spacing': ['warn', 'never'],
    'no-restricted-properties': [
      'error',
      {
        object: 'require',
        property: 'ensure',
        message:
          'Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting'
      },
      {
        object: 'System',
        property: 'import',
        message:
          'Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting'
      }
    ],
    'getter-return': 'warn',

    // Configure eslint-plugin-import
    'import/first': 'error',
    'import/no-amd': 'error',
    'import/no-webpack-loader-syntax': 'error',

    // Configure eslint-plugin-react
    'react/forbid-foreign-prop-types': [
      'warn',
      {
        allowInPropTypes: true
      }
    ],
    'react/jsx-no-comment-textnodes': 'warn',
    'react/jsx-no-duplicate-props': 'warn',
    'react/jsx-no-target-blank': 'warn',
    'react/jsx-no-undef': 'error',
    'react/jsx-pascal-case': [
      'warn',
      {
        allowAllCaps: true,
        ignore: []
      }
    ],
    'react/jsx-uses-react': 'warn',
    'react/jsx-uses-vars': 'warn',
    'react/no-children-prop': 'warn',
    'react/no-access-state-in-setstate': 'warn',
    'react/no-danger': 'error',
    'react/no-danger-with-children': 'warn',
    'react/no-direct-mutation-state': 'warn',
    'react/no-is-mounted': 'warn',
    'react/no-this-in-sfc': 'warn',
    'react/no-typos': 'error',
    'react/react-in-jsx-scope': 'error',
    'react/require-render-return': 'error',
    'react/style-prop-object': 'warn',
    'react/void-dom-elements-no-children': 'warn',

    // Configure eslint-plugin-jsx-a11y
    'jsx-a11y/accessible-emoji': 'warn',
    'jsx-a11y/alt-text': 'warn',
    'jsx-a11y/anchor-has-content': 'warn',
    'jsx-a11y/anchor-is-valid': [
      'warn',
      {
        aspects: ['noHref', 'invalidHref']
      }
    ],
    'jsx-a11y/aria-activedescendant-has-tabindex': 'warn',
    'jsx-a11y/aria-props': 'warn',
    'jsx-a11y/aria-proptypes': 'warn',
    'jsx-a11y/aria-role': [
      'warn',
      {
        ignoreNonDOM: true
      }
    ],
    'jsx-a11y/aria-unsupported-elements': 'warn',
    'jsx-a11y/heading-has-content': 'warn',
    'jsx-a11y/iframe-has-title': 'warn',
    'jsx-a11y/img-redundant-alt': 'warn',
    'jsx-a11y/no-access-key': 'warn',
    'jsx-a11y/no-distracting-elements': 'warn',
    'jsx-a11y/no-redundant-roles': 'warn',
    'jsx-a11y/role-has-required-aria-props': 'warn',
    'jsx-a11y/role-supports-aria-props': 'warn',
    'jsx-a11y/scope': 'warn',

    // Configure eslint-plugin-react-hooks
    'react-hooks/exhaustive-deps': 'warn',
    'react-hooks/rules-of-hooks': 'error'
  }
};

module.exports = config;
export default config;
