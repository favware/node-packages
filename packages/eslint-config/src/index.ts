const config = {
  root: true,
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.eslint.json',
    jsx: true,
    sourceType: 'module',
    ecmaVersion: 2019,
    ecmaFeatures: { jsx: true },
  },
  extends: [
    'plugin:jest/all',
    'eslint:recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended'
  ],
  env: {
    node: true,
    es6: true,
    jest: true,
    browser: true,
    commonjs: true,
  },
  rules: {

    // TypeScript
    '@typescript-eslint/indent': [ 'error', 2 ],
    '@typescript-eslint/no-non-null-assertion': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/prefer-interface': 'off',
    '@typescript-eslint/consistent-type-definitions': [
      'error',
      'type'
    ],
    '@typescript-eslint/await-thenable': 'error',
    '@typescript-eslint/ban-types': [
      'error',
      {
        types: {
          String: {
            message: 'Use string instead',
            fixWith: 'string',
          },
          Object: {
            message: 'Use object instead',
            fixWith: 'object',
          },
          '{}': {
            message: 'Use object instead',
            fixWith: 'object',
          },
          Array: null,
        },
      }
    ],
    '@typescript-eslint/func-call-spacing': [
      'error'
    ],
    '@typescript-eslint/member-ordering': 'error',
    '@typescript-eslint/no-empty-function': 'error',
    '@typescript-eslint/no-extra-parens': [
      'error',
      'functions'
    ],
    '@typescript-eslint/no-extraneous-class': [
      'error',
      {allowStaticOnly: true}
    ],
    '@typescript-eslint/no-for-in-array': 'error',
    '@typescript-eslint/no-require-imports': 'error',
    '@typescript-eslint/no-this-alias': 'error',
    '@typescript-eslint/no-unnecessary-qualifier': 'error',
    '@typescript-eslint/no-unnecessary-type-assertion': 'error',
    '@typescript-eslint/no-useless-constructor': 'error',
    '@typescript-eslint/prefer-for-of': 'error',
    '@typescript-eslint/prefer-includes': 'error',
    '@typescript-eslint/promise-function-async': 'off',
    '@typescript-eslint/require-array-sort-compare': 'error',
    '@typescript-eslint/restrict-plus-operands': 'off',
    '@typescript-eslint/semi': [
      'error',
      'always'
    ],
    '@typescript-eslint/unbound-method': 'error',
    '@typescript-eslint/unified-signatures': 'error',
    '@typescript-eslint/camelcase': [ 'error', { allow: [ 'api_key', 'app_id', 'show_alternative', 'store_link', 'grant_type' ] } ],

    // Jest Rules
    'jest/consistent-test-it': [ 'error', {fn: 'test'} ],
    'jest/no-hooks': 'off',
    'jest/lowercase-name': 'off',
    'jest/no-export': 'off',
    'jest/prefer-expect-assertions': 'off',
    'jest/no-if': 'off',
    'jest/prefer-inline-snapshots': 'off',
    'jest/expect-expect': [ 'error', { assertFunctionNames: [ 'expect', 'expectSaga', 'testSaga' ] } ],

    // Possible Errors
    'no-console': 'warn',
    'no-template-curly-in-string': 'error',
    'require-atomic-updates': 'off',

    // Best Practices
    'accessor-pairs': 'error',
    'array-callback-return': 'error',
    'block-scoped-var': 'error',
    'consistent-return': 'error',
    curly: [
      'error',
      'multi-line',
      'consistent'
    ],
    'default-case': 'error',
    'dot-location': [
      'error',
      'property'
    ],
    eqeqeq: [
      'error',
      'smart'
    ],
    'max-classes-per-file': [
      'error',
      3
    ],
    'no-alert': 'error',
    'no-caller': 'error',
    'no-div-regex': 'error',
    'no-else-return': 'error',
    'no-eq-null': 'error',
    'no-eval': 'error',
    'no-extend-native': 'error',
    'no-extra-bind': 'error',
    'no-extra-label': 'error',
    'no-floating-decimal': 'error',
    'no-implicit-coercion': 'error',
    'no-implicit-globals': 'error',
    'no-implied-eval': 'error',
    'no-invalid-this': 'error',
    'no-iterator': 'error',
    'no-lone-blocks': 'error',
    'no-loop-func': 'error',
    'no-multi-spaces': 'error',
    'no-multi-str': 'error',
    'no-new': 'error',
    'no-new-func': 'error',
    'no-new-wrappers': 'error',
    'no-octal-escape': 'error',
    'no-proto': 'error',
    'no-prototype-builtins': 'off',
    'no-return-assign': 'error',
    'no-return-await': 'error',
    'no-script-url': 'error',
    'no-self-compare': 'error',
    'no-sequences': 'error',
    'no-throw-literal': 'error',
    'no-unmodified-loop-condition': 'error',
    'no-unused-expressions': 'error',
    'no-useless-call': 'error',
    'no-useless-concat': 'error',
    'no-useless-return': 'error',
    'no-void': 'error',
    'no-warning-comments': [
      'warn',
      {
        terms: [
          'todo',
          'fixme'
        ],
        location: 'anywhere',
      }
    ],
    'prefer-promise-reject-errors': 'error',
    radix: [
      'error',
      'as-needed'
    ],
    'wrap-iife': 'error',
    yoda: 'error',

    // Strict Mode
    strict: [
      'error',
      'never'
    ],

    // Variables
    'no-label-var': 'error',
    'no-shadow': 'error',
    'no-undef-init': 'error',

    // Node.js and CommonJS
    'callback-return': 'error',
    'global-require': 'error',
    'handle-callback-err': 'error',
    'no-buffer-constructor': 'error',
    'no-mixed-requires': 'error',
    'no-new-require': 'error',
    'no-path-concat': 'error',

    // Stylistic Issues
    'array-bracket-spacing': [
      'error',
      'always'
    ],
    'block-spacing': 'error',
    'brace-style': 'error',
    'capitalized-comments': 'error',
    'comma-dangle': [
      'error',
      {
        arrays: 'never',
        objects: 'always-multiline',
        imports: 'never',
        exports: 'never',
        functions: 'ignore',
      }
    ],
    'comma-spacing': 'error',
    'comma-style': 'error',
    'computed-property-spacing': 'error',
    'eol-last': [
      'error',
      'never'
    ],
    'func-name-matching': 'error',
    'func-names': 'error',
    'func-style': [
      'error',
      'declaration',
      {allowArrowFunctions: true}
    ],
    'implicit-arrow-linebreak': 'error',
    'jsx-quotes': [
      'error',
      'prefer-single'
    ],
    'key-spacing': 'error',
    'keyword-spacing': 'error',
    'linebreak-style': 'warn',
    'lines-between-class-members': [
      'error',
      'always',
      {exceptAfterSingleLine: true}
    ],
    'max-len': [
      'warn',
      {
        code: 200,
        tabWidth: 2,
      }
    ],
    'max-statements-per-line': [
      'error',
      {max: 1}
    ],
    'new-cap': [ 'error', { capIsNewExceptionPattern: '^(?:[A-Z]+)_'} ],
    'new-parens': 'error',
    'no-array-constructor': 'error',
    'no-lonely-if': 'error',
    'no-mixed-operators': 'error',
    'no-multi-assign': 'error',
    'no-multiple-empty-lines': 'error',
    'no-nested-ternary': 'error',
    'no-new-object': 'error',
    'no-tabs': 'error',
    'no-trailing-spaces': 'error',
    'no-underscore-dangle': [
      'error',
      {enforceInMethodNames: true}
    ],
    'no-unneeded-ternary': 'error',
    'no-whitespace-before-property': 'error',
    'nonblock-statement-body-position': 'error',
    'object-curly-newline': [
      'error',
      {
        ObjectExpression: {
          multiline: true,
          minProperties: 4,
        },
        ObjectPattern: {
          multiline: true,
          minProperties: 4,
        },
        ImportDeclaration: {
          multiline: true,
          minProperties: 10,
        },
        ExportDeclaration: {
          multiline: true,
          minProperties: 4,
        },
      }
    ],
    'one-var': [
      'error',
      {
        separateRequires: true,
        var: 'never',
        let: 'never',
        const: 'never',
      }
    ],
    'one-var-declaration-per-line': [
      'error',
      'always'
    ],
    'operator-linebreak': [
      'error',
      'after',
      {
        overrides: {
          '?': 'ignore',
          ':': 'ignore',
        },
      }
    ],
    'padded-blocks': [
      'error',
      'never'
    ],
    'padding-line-between-statements': [
      'error',
      {
        blankLine: 'always',
        prev: '*',
        next: 'return',
      }
    ],
    'prefer-object-spread': 'error',
    'quote-props': [
      'error',
      'as-needed',
      {numbers: true}
    ],
    quotes: [
      'error',
      'single'
    ],
    'semi-spacing': [
      'error',
      {
        before: false,
        after: true,
      }
    ],
    'semi-style': [
      'error',
      'last'
    ],
    'sort-vars': 'error',
    'space-before-blocks': 'error',
    'space-before-function-paren': [
      'error',
      {
        anonymous: 'always',
        named: 'never',
        asyncArrow: 'always',
      }
    ],
    'space-in-parens': 'error',
    'space-infix-ops': 'error',
    'space-unary-ops': 'error',
    'switch-colon-spacing': 'error',
    'template-tag-spacing': 'error',
    'no-useless-computed-key': 'error',

    // ECMAScript 6
    'arrow-parens': [
      'error',
      'as-needed'
    ],
    'arrow-spacing': 'error',
    'generator-star-spacing': [
      'error',
      {
        before: false,
        after: true,
      }
    ],
    'no-duplicate-imports': 'error',
    'no-useless-rename': 'error',
    'prefer-arrow-callback': 'error',
    'prefer-numeric-literals': 'error',
    'prefer-template': 'error',
    'rest-spread-spacing': 'error',
    'template-curly-spacing': 'error',
    'yield-star-spacing': [
      'error',
      'after'
    ],
  },
};

module.exports = config;
export default config;