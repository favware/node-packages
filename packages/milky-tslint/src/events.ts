import StrictEventEmitter from 'strict-event-emitter-types';
import { EventEmitter } from 'events';
import { ITSLintFile } from './types';
import PluginError from 'plugin-error';

/**
 * @typdef MilkyEvents Events output by this library
 */
export type MilkyEvents = {
  /**
   * Fired where there is linted data
   */
  data: (file: ITSLintFile) => void;
  /**
   * Fired when an error occurs during linting
   */
  error: (err: PluginError) => void;
  /**
   * Fired when lintin is complete
   */
  end: () => void;
};

export const eventemitter: StrictEventEmitter<EventEmitter, MilkyEvents> = new EventEmitter();
export default eventemitter;