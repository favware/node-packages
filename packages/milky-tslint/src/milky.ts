/* eslint-disable no-console, @typescript-eslint/no-explicit-any */
import chalk from 'chalk';
import moment from 'moment';
import PluginError from 'plugin-error';
import through from 'through';
import TSLint from 'tslint';
import { RuleFailure } from 'tslint/lib/language/rule/rule';
import { IPluginOptions, LogLevels, LintFormatters, ITSLintFile, IReportOptions } from './types';
import eventemitter from './events';

// eslint-disable-next-line
const map = require('map-stream');

/**
 * Helper function to check if a value is a string
 * @param {any} value to check whether or not it is a string
 * @returns {boolean} Returns true if the value is a string
 * @ignore
 */
const isString = (value: any): boolean => {
  return typeof value === 'string';
};

/**
 * Returns the TSLint from the options, or if not set, the default TSLint.
 * @param {IPluginOptions} options
 * @returns {any} TSLint module
 * @ignore
 */
const getTslint = (options: IPluginOptions): any => {
  if (options && options.tslint) {
    return options.tslint;
  }

  return TSLint;
};

/**
 * Log an event or error
 * @param {string} message Message that should be logged
 * @param {LogLevels} [level] Logging level, one of {@link LogLevels}
 * @ignore
 */
const log = (message: string, level?: LogLevels) => {
  const prefix = `[${chalk.gray(moment().format('HH:mm:ss'))}] [${chalk.cyan('@favware/milky-tslint')}]`;

  switch (level) {
    case LogLevels.ERROR:
      return console.error(prefix, chalk.red(message));
    case LogLevels.INFO:
      return console.info(prefix, chalk.green(message));
    case LogLevels.WARN:
      return console.warn(prefix, chalk.yellow(message));
    default:
      return console.log(prefix, message);
  }
};

let unscopedPluginOptions: IPluginOptions;

/**
 * @method milkyLint
 * @description TSLint plugin for Gulp
 * @param pluginOptions Options to pass to the plugin
 */
export const milkyLint = (pluginOptions: IPluginOptions = {}) => {
  unscopedPluginOptions = pluginOptions;

  const options: TSLint.ILinterOptions = {
    fix: pluginOptions.fix || false,
    formatter: pluginOptions.formatter || LintFormatters.STYLISH,
    formattersDirectory: pluginOptions.formattersDirectory || undefined,
    quiet: pluginOptions.quiet || false,
    rulesDirectory: pluginOptions.rulesDirectory || undefined,
  };

  const linter = getTslint(pluginOptions);
  const tslint = new linter.Linter(options, pluginOptions.program);

  return map((file: ITSLintFile, cb: (error: any, file?: ITSLintFile) => void) => {
    if (file.isNull()) return cb(null, file);
    if (file.isStream()) return cb(new PluginError('@favware/milky-tslint', 'streaming is not supported'));

    const configuration = !pluginOptions.configuration || isString(pluginOptions.configuration)
      ? linter.Configuration.findConfiguration(pluginOptions.configuration || null, file.path).results
      : pluginOptions.configuration;

    tslint.lint(file.path, file.contents.toString(), configuration);
    file.tslint = tslint.getResult();

    // Clear all results for current file from tslint
    tslint.failures = [];
    tslint.fixes = [];

    return cb(null, file);
  });
};

/**
 * @method milkyReport
 * @description Reporter for milky-tslint
 * @param options Reporter options
 */
export const milkyReport = (options: IReportOptions = {
  emitError: false, reportLimit: 0, summarizeFailureOutput: true, allowWarnings: false,
}) => {
  if (!options.emitError) options.emitError = false;
  if (!options.reportLimit) options.reportLimit = 0;
  if (!options.summarizeFailureOutput) options.summarizeFailureOutput = true;
  if (!options.allowWarnings) options.allowWarnings = false;

  const errorFiles: ITSLintFile[] = [];
  let allFailures: RuleFailure[] = [];

  let totalReported = 0;

  const reportFailures = (file: ITSLintFile) => {
    if (file.tslint) {
      let errorCount = file.tslint.errorCount;
      if (!options.allowWarnings) errorCount += file.tslint.warningCount;

      if (errorCount > 0) {
        errorFiles.push(file);
        allFailures = allFailures.concat(file.tslint.failures);

        if (options.reportLimit !== undefined && (options.reportLimit <= 0 || (options.reportLimit && options.reportLimit > totalReported))) {
          if (file.tslint.output !== undefined) console.log(file.tslint.output);

          totalReported += errorCount;

          if (options.reportLimit > 0 && options.reportLimit <= totalReported) {
            log(`More than ${options.reportLimit} failures reported. Turning off reporter.`, LogLevels.WARN);
          }
        }
      } else if (options.allowWarnings && file.tslint.warningCount > 0) {
        // Íf only warnings were emitted, format and print them
        const FormatterConstructor = TSLint.findFormatter(unscopedPluginOptions.formatter!)!;
        const formatter = new FormatterConstructor();

        const warnings = (file.tslint as TSLint.LintResult).failures.filter(failure => failure.getRuleSeverity() === 'warning');
        log(formatter.format(warnings), LogLevels.WARN);
      }
    }

    milkyReport.eventEmitter.emit('data', file);
  };

  const throwErrors = () => {
    if (options && errorFiles.length > 0) {
      let failureOutput = allFailures;
      let ignoreFailureCount = 0;

      if (options.reportLimit !== undefined && options.reportLimit > 0) {
        ignoreFailureCount = allFailures.length - options.reportLimit;
        failureOutput = allFailures.slice(0, options.reportLimit);
      }

      let errorOutput = 'Failed to lint: ';

      errorOutput += options.summarizeFailureOutput ? `${failureOutput.length} errors.` : `${failureOutput}.`;

      if (ignoreFailureCount) errorOutput += ` (${ignoreFailureCount} other errors not shown.)`;

      if (options.emitError) {
        return milkyReport.eventEmitter.emit('error', new PluginError('@favware/milky-tslint', errorOutput));
      } else if (options.summarizeFailureOutput) {
        return log(errorOutput, LogLevels.WARN);
      }
    } else if (options && options.summarizeFailureOutput) {
      return log('🎉  Awesome, your linting is Milky Clean! Get yourself a glass of 🥛  for the awesome work!', LogLevels.INFO);
    }

    return milkyReport.eventEmitter.emit('end');
  };

  return through(reportFailures, throwErrors);
};

milkyReport.eventEmitter = eventemitter;

export default milkyLint;