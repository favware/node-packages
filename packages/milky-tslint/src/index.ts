import { milkyLint } from './milky';

export * from './milky';
export default milkyLint;