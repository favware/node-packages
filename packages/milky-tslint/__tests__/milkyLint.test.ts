import { milkyLint } from '../src';

describe('Milky Lint', () => {
  test('should match snapshot', () => {
    expect(milkyLint).toMatchSnapshot();
  });
});