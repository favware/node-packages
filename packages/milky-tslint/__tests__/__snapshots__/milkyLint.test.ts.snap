// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Milky Lint should match snapshot 1`] = `[Function]`;
